bazel/copts.bzl

```diff
@@ -1,4 +1,11 @@
-COPTS = select({
-    "@bazel_tools//tools/cpp:msvc": ["/std:c++17", "/permissive-"],
-    "//conditions:default": ["-std=c++17"],
-})
+load("@bazel_skylib//lib:selects.bzl", "selects")
+
+COPTS = selects.with_or({
+    ("@rules_cc//cc/compiler:clang", "@rules_cc//cc/compiler:gcc", "@rules_cc//cc/compiler:mingw-gcc"): [
+        "-std=c++17",
+    ],
+    ("@rules_cc//cc/compiler:msvc-cl", "@rules_cc//cc/compiler:clang-cl"): [
+        "/std:c++17",
+        "/permissive-",
+    ],
```

Comment:
This is from some feedback I got while submitting to the bazel central registry https://github.com/bazelbuild/bazel-central-registry/pull/811#discussion_r1284035470